doc/i2p.md

```diff
@@ -47,9 +53,34 @@ In a typical situation, this suffices:
 bitcoind -i2psam=127.0.0.1:7656
 ```
 
-The first time Bitcoin Core connects to the I2P router, its I2P address (and
-corresponding private key) will be automatically generated and saved in a file
-named `i2p_private_key` in the Bitcoin Core data directory.
+The first time Bitcoin Core connects to the I2P router, if a persistent I2P
+address is to be used, then it (and its corresponding private key) will be
+automatically generated and saved in a file named `i2p_private_key` in the
+Bitcoin Core data directory. A persistent I2P address is used for accepting
+incoming connections (`-i2pacceptincoming`) and for making outgoing connections
+if `-i2ptransientout` is not used.
+
+## Persistent vs transient I2P addresses
+
+In I2P connections, the host that receives the connection knows the I2P address
+of the connection initiator. This is unlike the Tor network where the recipient
+does not know who is connecting to them, not even the initiator's Tor address.
+
+So, even if an I2P node is not accepting incoming connections, they are known
+to other nodes by their outgoing I2P address. This creates an opportunity to
+white-list given nodes or treat them differently based on their I2P address.
+However, this also creates an opportunity to fingerprint or analyze a given node
+because it always uses the same I2P address when it connects to other nodes. If
+this is undesirable, then a node operator can use `-i2ptransientout` to generate
+a transient (disposable), one-time I2P address for each new outgoing connection.
+That address is never going to be reused again, not even if reconnecting to the
+same peer later.
+
+`-i2ptransientout` only affects addresses for outgoing connections and not the
+address for accepting incoming connections (if `-i2pacceptincoming` is given).
+This is because the address for accepting connections is advertised to the
+Bitcoin network and it takes time for it to propagate and be known by other
+nodes. So, it does not make sense to change it often - it must be persistent.
```

Comment:
suggestions

```diff
-In I2P connections, the host that receives the connection knows the I2P address
-of the connection initiator. This is unlike the Tor network where the recipient
-does not know who is connecting to them, not even the initiator's Tor address.
-
-So, even if an I2P node is not accepting incoming connections, they are known
-to other nodes by their outgoing I2P address. This creates an opportunity to
-white-list given nodes or treat them differently based on their I2P address.
-However, this also creates an opportunity to fingerprint or analyze a given node
-because it always uses the same I2P address when it connects to other nodes. If
-this is undesirable, then a node operator can use `-i2ptransientout` to generate
-a transient (disposable), one-time I2P address for each new outgoing connection.
-That address is never going to be reused again, not even if reconnecting to the
-same peer later.
-
-`-i2ptransientout` only affects addresses for outgoing connections and not the
-address for accepting incoming connections (if `-i2pacceptincoming` is given).
-This is because the address for accepting connections is advertised to the
-Bitcoin network and it takes time for it to propagate and be known by other
-nodes. So, it does not make sense to change it often - it must be persistent.
+With I2P connections, the connection receiver sees the I2P address of the
+connection initiator, unlike the Tor network where the receiver does
+not know who is connecting to them, not even the initiator's Tor address.
+
+Even if an I2P node does not accept incoming connections, it is known to other
+nodes by its outgoing I2P address. If this address is always the same when it
+connects to other nodes, this creates an opportunity to discriminate,
+fingerprint or analyze it based on its I2P address.
+
+For this reason, the `-i2ptransientout` configuration option generates and
+announces a new, transient I2P address for each new outbound connection. The
+address will never be reused, not even if reconnecting to the same peer later.
+
+`-i2ptransientout` only affects addresses announced to outbound connections and
+not the address for accepting inbound connections (provided that
+`-i2pacceptincoming` is on).  This is done because the address for accepting
+connections is advertised to the Bitcoin network and it takes time for it to
+propagate and be known by other nodes. It does not make sense to change it
+often -- it must be persistent.
```

Reply:
Took some of this.